attest/verifier/src/status.rs

```diff
@@ -647,19 +927,111 @@ mod test {
         };
 
         let data = VerificationReportData::try_from(&report).expect("Could not parse IAS result");
-        assert!(verifier.verify(&data))
+        assert!(!verifier.verify(&data))
+    }
+
+    /// Ensure a CONFIGURATION_AND_SW_HARDENING_NEEDED result with the expected
+    /// MRSIGNER, product, and minimum version fails if the advisory isn't
+    /// accounted for as both config and sw.
+    #[test]
+    fn mrsigner_fail_multi_config_sw_no_sw() {
+        let verifier = MrSignerVerifier {
+            mr_signer: MrSigner::from(&MR_SIGNER),
+            product_id: 0,
+            minimum_svn: 0,
+            config_ids: vec!["INTEL-SA-00334".to_owned(), "INTEL-SA-00615".to_owned()],
+            sw_ids: vec![],
+        };
+
+        let report = VerificationReport {
+            sig: Default::default(),
+            chain: vec![],
+            http_body: IAS_CONFIG_SW.trim().to_owned(),
+        };
+
+        let data = VerificationReportData::try_from(&report).expect("Could not parse IAS result");
+        assert!(!verifier.verify(&data))
+    }
+
+    /// Ensure a CONFIGURATION_AND_SW_HARDENING_NEEDED result with the expected
+    /// MRSIGNER, product, and minimum version fails if the advisory isn't
+    /// accounted for in both config and sw.
+    #[test]
+    fn mrsigner_fail_multi_config_sw_short_sw() {
+        let verifier = MrSignerVerifier {
+            mr_signer: MrSigner::from(&MR_SIGNER),
+            product_id: 0,
+            minimum_svn: 0,
+            config_ids: vec!["INTEL-SA-00615".to_owned()],
+            sw_ids: vec![],
+        };
+
+        let report = VerificationReport {
+            sig: Default::default(),
+            chain: vec![],
+            http_body: IAS_CONFIG_SW.trim().to_owned(),
+        };
+
+        let data = VerificationReportData::try_from(&report).expect("Could not parse IAS result");
+        assert!(!verifier.verify(&data))
+    }
+
```

Comment:
I'm not sure this test is needed, based on the setup this seems equivalent to the testcase on line 1007.  Perhaps this was meant to test having a config advisory but the advisory in the report is different.

Reply:
This is testing what happens when config is empty, the other is testing when config has one element but needs two.